crates/driver/src/infra/api/mod.rs

```diff
@@ -58,7 +58,16 @@ impl Api {
         app = routes::metrics(app);
         app = routes::healthz(app);
 
-        let metrics_bad_token_detector = bad_tokens::metrics::Detector::default();
+        let simulation_bad_token_detector = bad_tokens::simulation::Detector::new(
+            self.bad_token_detection.simulation_bad_token_max_age,
+            &self.eth,
+        );
+
+        let metrics_bad_token_detector = bad_tokens::metrics::Detector::new(
+            self.bad_token_detection.metrics_bad_token_failure_ratio,
+            self.bad_token_detection
+                .metrics_bad_token_required_measurements,
+        );
```

Comment:
Different solver instances must not share the same metrics based detector all of them need to have their own.
The reason they should share the simulation detector is that it's predictable. All solvers running the simulation detector should come to the same conclusions about all tested tokens.
The metrics based approach is unique to each solver. Some solvers do support trading fee on transfer tokens. For those solvers the failure measurements for fee-on-transfer tokens will be significantly less than on solvers that don't support them correctly.

Reply:
Got it, makes sense, thanks. Fixed this.